sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -967,7 +975,7 @@ static void _handle_nbr_sol(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
     }
 #endif  /* CONFIG_GNRC_IPV6_NIB_SLAAC */
     if (ipv6_addr_is_unspecified(&ipv6->src)) {
-        gnrc_ndp_nbr_adv_send(&nbr_sol->tgt, netif, &ipv6->src, false, NULL);
+        gnrc_ndp_nbr_adv_send(&nbr_sol->tgt, tgt_netif, &ipv6->src, false, NULL);
```

Comment:
From what I understand If the neighbor is solicited via `netif` corresponds interface A in your description and `tgt_netif` interface B. Shouldn't the neighbor advertisement go out the same interface (`netif`) the solicitation was received on?

Reply:
Hm this works because the source is a global address so the interface is ignored for sending.
When I set the 'correct' interface [`gnrc_ndp_nbr_adv_send()`](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ndp/gnrc_ndp.c#L335) will bail out.

```
2021-06-18 17:50:05,264 # Iface  7  HWaddr: 5A:84:3B:4F:A9:03 
2021-06-18 17:50:05,266 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-18 17:50:05,267 #           Source address length: 6
2021-06-18 17:50:05,268 #           Link type: wired
2021-06-18 17:50:05,269 #           inet6 addr: fe80::5884:3bff:fe4f:a903  scope: link  VAL
2021-06-18 17:50:05,271 #           inet6 addr: 2001:16b8:45f9:73f8:5884:3bff:fe4f:a903  scope: global  VAL
2021-06-18 17:50:05,272 #           inet6 group: ff02::2
2021-06-18 17:50:05,273 #           inet6 group: ff02::1
2021-06-18 17:50:05,274 #           inet6 group: ff02::1:ff4f:a903
2021-06-18 17:50:05,274 #           
2021-06-18 17:50:05,275 #           Statistics for Layer 2
2021-06-18 17:50:05,276 #             RX packets 7  bytes 626
2021-06-18 17:50:05,277 #             TX packets 5 (Multicast: 5)  bytes 438
2021-06-18 17:50:05,278 #             TX succeeded 0 errors 0
2021-06-18 17:50:05,279 #           Statistics for IPv6
2021-06-18 17:50:05,279 #             RX packets 7  bytes 528
2021-06-18 17:50:05,280 #             TX packets 5 (Multicast: 5)  bytes 368
2021-06-18 17:50:05,280 #             TX succeeded 5 errors 0
2021-06-18 17:50:05,280 # 
2021-06-18 17:50:05,281 # Iface  6  HWaddr: 5A:84:3B:4F:A9:02 
2021-06-18 17:50:05,281 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2021-06-18 17:50:05,282 #           Source address length: 6
2021-06-18 17:50:05,282 #           Link type: wired
2021-06-18 17:50:05,283 #           inet6 addr: fe80::5884:3bff:fe4f:a902  scope: link  VAL
2021-06-18 17:50:05,284 #           inet6 addr: 2001:16b8:45f9:73fa:5884:3bff:fe4f:a902  scope: global  VAL
2021-06-18 17:50:05,284 #           inet6 group: ff02::2
2021-06-18 17:50:05,285 #           inet6 group: ff02::1
2021-06-18 17:50:05,285 #           inet6 group: ff02::1:ff4f:a902
2021-06-18 17:50:05,285 #           
2021-06-18 17:50:05,286 #           Statistics for Layer 2
2021-06-18 17:50:05,286 #             RX packets 5  bytes 805
2021-06-18 17:50:05,287 #             TX packets 13 (Multicast: 13)  bytes 1206
2021-06-18 17:50:05,287 #             TX succeeded 0 errors 0
2021-06-18 17:50:05,288 #           Statistics for IPv6
2021-06-18 17:50:05,288 #             RX packets 5  bytes 735
2021-06-18 17:50:05,289 #             TX packets 13 (Multicast: 13)  bytes 1024
2021-06-18 17:50:05,289 #             TX succeeded 13 errors 0
2021-06-18 17:50:05,289 # 
2021-06-18 17:50:22,014 # nib: Handle packet (icmpv6->type = 135)
2021-06-18 17:50:22,015 # nib: Received valid neighbor solicitation on 7:
2021-06-18 17:50:22,016 #      - Target address: 2001:16b8:45f9:73fa:5884:3bff:fe4f:a902
2021-06-18 17:50:22,016 #      - Source address: 2001:16b8:45f9:73f8:a4ed:29ff:feec:d4f3
2021-06-18 17:50:22,017 #      - Destination address: ff02::1:ff4f:a902
2021-06-18 17:50:22,018 # ndp: send neighbor advertisement (iface: 6, tgt: 2001:16b8:45f9:73fa:5884:3bff:fe4f:a902, dst: 2001:16b8:45f9:73f8:a4ed:29ff:feec:d4f3, supply_tl2a: 1)
2021-06-18 17:50:22,019 # ndp: building target link-layer address option (l2addr: 5A:84:3B:4F:A9:02)
2021-06-18 17:50:22,020 # ndp: building neighbor advertisement message
2021-06-18 17:50:22,021 # nib: get next hop link-layer address of 2001:16b8:45f9:73f8:a4ed:29ff:feec:d4f3%6
2021-06-18 17:50:22,022 # nib: 2001:16b8:45f9:73f8:a4ed:29ff:feec:d4f3 is on-link or in NC, start address resolution
2021-06-18 17:50:22,023 # nib: resolve address 2001:16b8:45f9:73f8:a4ed:29ff:feec:d4f3%7 from neighbor cache
```

But I agree that it's pretty hacky to rely on this - let's adapt `gnrc_ndp_nbr_adv_send()` instead.